### PR TITLE
[CANNOLI-55] Fix compile errors due to upstream commit 0afbda96.

### LINKED ISSUE #55: Compile errors with latest ADAM 0.23.0-SNAPSHOT version
```
...
[ERROR] cannoli/src/main/scala/org/bdgenomics/cannoli/Bwa.scala:199: error: value copy
is not a member of org.bdgenomics.adam.rdd.read.AlignmentRecordRDD
[ERROR] possible cause: maybe a semicolon is missing before `value copy'?
[ERROR]      .copy(recordGroups = RecordGroupDictionary(Seq(RecordGroup(sample, sample))))
[ERROR]       ^
[ERROR] cannoli/src/main/scala/org/bdgenomics/cannoli/Bwa.scala:203: error: value copy
is not a member of org.bdgenomics.adam.rdd.read.AlignmentRecordRDD
[ERROR]      output.copy(sequences = sequences)
[ERROR]             ^
[ERROR] cannoli/src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala:32:
error: class FragmentRDD is abstract; cannot be instantiated
[ERROR]    new FragmentRDD(apply(rdd.rdd),
[ERROR]    ^
[ERROR] three errors found
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #55

### CODE REVIEW

File: src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala [hunk 1]
```diff
@@ -29,9 +29,8 @@ private[cannoli] object QuerynameGrouper extends Serializable {
   }
 
   def apply(rdd: AlignmentRecordRDD): FragmentRDD = {
-    new FragmentRDD(apply(rdd.rdd),
+    FragmentRDD(apply(rdd.rdd),
```
Comment (fnothaft) [inline src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala]: If you can wait for https://github.com/bigdatagenomics/adam/pull/1598, we can use `transmute` instead.

Comment (heuermh) [inline src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala]: Cannoli head is broken (and I get a dozen emails from Jenkins every morning to prove it), so I'd rather get this in first.

Comment (fnothaft) [inline src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala]: Ah yes, but what I'm getting at is that we're waiting for you to review/merge bigdatagenomics/adam#1598. Then, all the broken code can go away. ;)

Comment (heuermh) [inline src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala]: I know ... but that pull request is very interesting and has me thinking.  This change has +/- 19 LOC, that one has +/- 2824 LOC.

Comment (fnothaft) [inline src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala]: OK

**Comment by fnothaft** [comment]
Thanks @heuermh!

**Pull request merged**